src/lib/dapp/index.ts

```diff
@@ -112,31 +116,45 @@ export class DAppConnector {
     })
   }
 
+  /**
+   * Sets the logging level for the DAppConnector
+   * @param level - The logging level to set
+   */
+  public setLogLevel(level: 'error' | 'warn' | 'info' | 'debug'): void {
+    if (this.logger instanceof DefaultLogger) {
+      this.logger.setLogLevel(level)
+    }
+  }
+
   /**
    * Initializes the DAppConnector instance.
    * @param logger - `BaseLogger` for logging purposes (optional).
    */
-  async init({ logger }: { logger?: BaseLogger } = {}) {
+  async init({ logger }: { logger?: BaseLogger } = {}): Promise<void> {
     try {
       this.isInitializing = true
       if (!this.projectId) {
         throw new Error('Project ID is not defined')
       }
-      this.walletConnectClient = await SignClient.init({
+      const signClient = await SignClient.init({
         logger,
         relayUrl: 'wss://relay.walletconnect.com',
         projectId: this.projectId,
         metadata: this.dAppMetadata,
       })
+      this.walletConnectClient = signClient
       const existingSessions = this.walletConnectClient.session.getAll()
 
-      if (existingSessions.length > 0)
+      if (existingSessions.length > 0) {
         this.signers = existingSessions.flatMap((session) => this.createSigners(session))
-      else this.checkIframeConnect()
+      } else {
+        await this.checkIframeConnect()
+      }
```

Comment:
I agree with improving code style and formatting, but these changes should be in a separate commit to avoid any confusion with the purpose of this one.